library/src/updater.rs

```diff
@@ -367,7 +393,22 @@ pub fn report_launch_failure() -> anyhow::Result<()> {
                 )))?;
         // Ignore the error here, we'll try to activate the next best patch
         // even if we fail to mark this one as bad (because it was already bad).
-        let _ = state.mark_patch_as_bad(patch.number);
+        let mark_result = state.mark_patch_as_bad(patch.number);
```

Comment:
Same Q here re: saving state

Reply:
It gets accidentally saved below by the queue_patch call.

Reply:
Sorry by the activate_latest_bootable_patch() call.

Reply:
I added a doc about where the save happens.  Would you like others?  Happy to, just would like inspiration as to what docs you're seeking.